library/dnet.c

```diff
@@ -791,9 +791,6 @@ int dnet_server_send_write(struct dnet_server_send_ctl *send,
 	dnet_trans_create_send_all(s, &ctl);
 	dnet_session_destroy(s);
 
-	if (!err)
-		err = send->write_error;
-
```

Comment:
So, iterator's callback (`dnet_iterator_callback_server_send`) should check `write_error` too and should stop iterator if it isn't 0. I will make corresponding changes.